frontend/src/app/actions/newsletter-form.js

```diff
@@ -0,0 +1,29 @@
+import { createActions } from 'redux-actions';
+
+const actions = createActions(
+  {
+    newsletterFormSetEmail: email => email,
+    newsletterFormSetPrivacy: privacy => privacy
+  },
+  'newsletterFormSetFailed',
+  'newsletterFormSetSubmitting',
+  'newsletterFormSetSucceeded'
+);
+
+const subscribeActions = createActions({
+  newsletterFormSubscribe: (dispatch, email) => {
```

Comment:
Yeah, I don't really like actions dispatching other actions like this. 

Maybe the form should dispatch the `newsletterFormSetSubmitting` and `newsletterFormSubscribe` actions one after the other itself?

And then you can just yield an appropriate action payload describing success/fail at the end of the fetch without dispatching further events. [We kind of do this in `fetchExperiments`](https://github.com/mozilla/testpilot/blob/master/frontend/src/app/actions/experiments.js#L7), but sadly without good error handling. 

But, this payload could just be for one single action (`newsletterFormSubscribe`) that the reducer handles appropriately based on whether it has a `statusSuccess==true` or `statusFailed==true` property - or something like this.